Remap netcoreapp to netcoreapp3.0

weshaggard opened the pull request:
This PR builds on top of https://github.com/dotnet/corefx/pull/31993.

Addresses https://github.com/dotnet/corefx/issues/29384. 

PTAL @ericstj @joperezr @ViktorHofer It is probably easier to review commit-by-commit.

bartonjs commented on src/System.Security.Cryptography.Cng/pkg/System.Security.Cryptography.Cng.pkgproj:
```diff
@@ -17,6 +17,7 @@
     <HarvestIncludePaths Include="ref/netstandard1.4;runtimes/win/lib/netstandard1.4" />
     <HarvestIncludePaths Include="ref/netstandard1.6;runtimes/win/lib/netstandard1.6;lib/netstandard1.6" />
     <HarvestIncludePaths Include="ref/netcoreapp2.0;runtimes/win/lib/netcoreapp2.0" />
+    <HarvestIncludePaths Include="ref/netcoreapp2.1;lib/netcoreapp2.1;runtimes/win/lib/netcoreapp2.1" />
```
That sounds sensible.  I don't really know of a reason why I'd want to keep live-building 2.1.

weshaggard replied:
The only reason you might consider doing it is to avoid having the netcoreapp2.1 and netcoreapp3.0 assets in your package. We are essentially adding 2 netcoreapp targets in your package now.

jkotas commented on src/System.Threading.Tasks.Extensions/ref/Configurations.props:
```diff
@@ -7,6 +7,7 @@
     </PackageConfigurations>
     <BuildConfigurations>
       $(PackageConfigurations);
+      _netcoreapp2.1;
```
Why is with `underscore` prefix?

weshaggard replied:
The underscore makes this a placeholder configuration. So while building all configurations if it sees that configuration it will skip building anything for that configuration. This is needed in this case to prevent the netstandard configuration from being bin-placed into the netcoreapp2.1 ref folder during the ref build. Because for netcoreapp2.1 some of the APIs were moved down into System.Runtime and conflict with the APIs in the netstandard ref of this library.

ViktorHofer commented on src/System.IO.Pipes.AccessControl/src/System.IO.Pipes.AccessControl.csproj:
```diff
@@ -2,29 +2,11 @@
-    <IsPartialFacadeAssembly Condition="'$(TargetGroup)' != 'netstandard'">true</IsPartialFacadeAssembly>
+    <IncludeDefaultReferences>false</IncludeDefaultReferences>
+    <IsPartialFacadeAssembly>true</IsPartialFacadeAssembly>
     <Configurations>net461-Windows_NT-Debug;net461-Windows_NT-Release;netcoreapp-Windows_NT-Debug;netcoreapp-Windows_NT-Release;netfx-Windows_NT-Debug;netfx-Windows_NT-Release;netstandard-Debug;netstandard-Release;netstandard-Windows_NT-Debug;netstandard-Windows_NT-Release;uap-Windows_NT-Debug;uap-Windows_NT-Release</Configurations>
```
sync Configurations with Configurations.props

ViktorHofer replied:
I quickly spoke with @eerhardt. He mentioned that you were trying to combine the Configurations.props with the Configurations prop here. Do we have a tracking issue for that?

weshaggard replied:
Thanks for the reminder as I planned to do that all at once for all the projects.

weshaggard replied:
>I quickly spoke with @eerhardt. He mentioned that you were trying to combine the Configurations.props with the Configurations prop here. Do we have a tracking issue for that?

Not there isn't a tracking issue but I did make some attempts to figure out a good way to take the configurations in configurations.props and add the "Debug" and "Release" to them but my attempts have failed. So for now I think we will stick with the update task but I may make another attempt once we get our configuration system ported to arcade.

ViktorHofer replied:
ok thanks for the update

ViktorHofer commented on src/System.IO.Pipes.AccessControl/ref/System.IO.Pipes.AccessControl.csproj:
```diff
@@ -1,17 +1,12 @@
   <PropertyGroup>
     <ProjectGuid>{994DCE47-4CF6-479D-AB1D-4EA6A2809C34}</ProjectGuid>
-    <IsPartialFacadeAssembly Condition="'$(TargetsNetFx)' == 'true'">true</IsPartialFacadeAssembly>
     <Configurations>net461-Debug;net461-Release;netfx-Debug;netfx-Release;netstandard-Debug;netstandard-Release</Configurations>
```
same here

ViktorHofer commented on tools-local/targetgroups.props:
```diff
@@ -189,33 +189,46 @@
-    <TargetGroups Include="netcoreapp2.1aot">
-      <TargetFramework>netcoreapp2.1</TargetFramework>
+    <!-- netcoreapp is an alias for netcoreapp3.0 -->
+    <TargetGroups Include="netcoreapp">
```
We probably don't want to also do the netstandard mapping at this point?

weshaggard replied:
There is no new netstandard beyond 2.0 yet so everything still maps to that.

joperezr commented on Directory.Build.targets:
```diff
@@ -142,6 +142,14 @@
   <Import Condition="'$(IncludeVSTestReferences)' == 'true' AND ('$(BuildingNETCoreAppVertical)' == 'true' OR '$(BuildingNETFxVertical)' == 'true')" Project="$(PackagesDir)\microsoft.codecoverage\$(MicrosoftDotNetTestSdkVersion)\build\netstandard1.0\Microsoft.CodeCoverage.targets" />
 
+  <ItemGroup>
+    <!-- Should be added to BuildTools but for now adding here to enable netcoreapp3.0 support -->
```
NIT: Can we add an issue number to the comment so that the work gets tracked?

weshaggard replied:
Better I submitted the PR. https://github.com/dotnet/buildtools/pull/2151

joperezr commented on tools-local/targetgroups.props:
```diff
@@ -189,33 +189,46 @@
-      <NuGetTargetMoniker>.NETCoreApp,Version=v1.2</NuGetTargetMoniker>
-      <NuGetTargetMonikerShort>netcoreapp1.2</NuGetTargetMonikerShort>
+    <TargetGroups Include="netcoreapp3.0">
+      <NETCoreAppMaximumVersion>3.0</NETCoreAppMaximumVersion>
```
Question: What are you using this for?

weshaggard replied:
This is needed to build our projects now given that the CLI we are using doesn't support this 3.0 as a target framework so we need to set the max version to be able to build.

joperezr commented on src/System.Security.Cryptography.OpenSsl/pkg/System.Security.Cryptography.OpenSsl.pkgproj:
```diff
@@ -3,12 +3,12 @@
   <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), Directory.Build.props))\Directory.Build.props" />
   <ItemGroup>
     <ProjectReference Include="..\ref\System.Security.Cryptography.OpenSsl.csproj">
-      <!-- Removing net461 and AllXamarianFrameworks from validation because of https://github.com/dotnet/corefx/issues/25924 -->
```
Do we not need the xamarin framework validation to be added eventually anymore?

weshaggard replied:
I don't think so given that this package doesn't have any xamarin assets and consumes need to consume the binary from this package in order to use it there.

weshaggard commented:
test UWP CoreCLR x64 Debug Build

joperezr commented:
@weshaggard I just saw that as part of this you also removed the netstandard configuration from the System.IO.Pipes.AccessControl tests. Given that we now have this push to try to get a netstandard test suite available for other frameworks to validate functionality, should we add it back?

weshaggard commented:
>@weshaggard I just saw that as part of this you also removed the netstandard configuration from the System.IO.Pipes.AccessControl tests. Given that we now have this push to try to get a netstandard test suite available for other frameworks to validate functionality, should we add it back?

The reason I removed it was because we are no longer building the netfx configuration only uap/netcoreapp, but with the netstandard test configuration it was trying to run the tests on netfx which failed because it didn't have the binary necessary to run the tests. I don't have an issue adding the netstandard configuration back, although I think it is less important for this library because it isn't part of the standard. If you do add it back you will need to figure out something for the other test runs beyond netcoreapp/uap.
